src/NzbDrone.Core/Datastore/Migration/204_add_release_hash.cs

```diff
@@ -0,0 +1,75 @@
+using System.Collections.Generic;
+using System.Data;
+using System.IO;
+using Dapper;
+using FluentMigrator;
+using NzbDrone.Common.Extensions;
+using NzbDrone.Core.Datastore.Migration.Framework;
+
+namespace NzbDrone.Core.Datastore.Migration
+{
+    [Migration(204)]
+    public class add_add_release_hash : NzbDroneMigrationBase
+    {
+        protected override void MainDbUpgrade()
+        {
+            Alter.Table("EpisodeFiles").AddColumn("ReleaseHash").AsString().Nullable();
+
+            Execute.WithConnection(UpdateEpisodeFiles);
+        }
+
+        private void UpdateEpisodeFiles(IDbConnection conn, IDbTransaction tran)
+        {
+            var updates = new List<object>();
+
+            using (var cmd = conn.CreateCommand())
+            {
+                cmd.Transaction = tran;
+                cmd.CommandText = "SELECT \"Id\", \"SceneName\", \"RelativePath\", \"OriginalFilePath\" FROM \"EpisodeFiles\" WHERE \"ReleaseHash\" IS NULL";
+
+                using var reader = cmd.ExecuteReader();
+                while (reader.Read())
+                {
+                    var id = reader.GetInt32(0);
+                    var sceneName = reader.IsDBNull(1) ? null : reader.GetString(1);
+                    var relativePath = reader.IsDBNull(2) ? null : reader.GetString(2);
+                    var originalFilePath = reader.IsDBNull(3) ? null : reader.GetString(3);
+
+                    string title = null;
+                    if (sceneName.IsNotNullOrWhiteSpace())
+                    {
+                        title = sceneName;
+                    }
+                    else if (relativePath.IsNotNullOrWhiteSpace())
+                    {
+                        title = Path.GetFileNameWithoutExtension(relativePath);
+                    }
+                    else if (originalFilePath.IsNotNullOrWhiteSpace())
+                    {
+                        title = Path.GetFileNameWithoutExtension(originalFilePath);
+                    }
+
+                    if (!title.IsNullOrWhiteSpace())
```

Comment:
Since the relative path probably doesn't have the hash (given it was only available via a Preferred Word match), I think this should be two tiered.

1. Check the scene name or original file name, giving preference to the scene name as above if either is available
2. Check the existing file name, which will cover cases where the original information is unavailable, but the file name does have it.

Something like: 

```
ParsedEpisodeInfo parsedEpisodeInfo = null;

var originalTitle = sceneName ?? originalFilePath.IsNotNullOrWhiteSpace() ? Path.GetFileNameWithoutExtension(originalFilePath) : null;

if (originalTitle.IsNotNullOrWhiteSpace())
{
  parsedEpisodeInfo = Parser.Parser.ParseTitle(originalTitle);
}

if ( parsedEpisodeInfo?.ReleaseHash.IsNullOrWhiteSpace())
{
  parseEpisodeInfo = Parser.Parser.ParseTitle(Path.GetFileNameWithoutExtension(relativePath));
}

if (parsedEpisodeInfo?.ReleaseHash.IsNotNullOrWhiteSpace())
{
  updates.Add(new
	{
	    Id = id,
	    ReleaseHash = parsedEpisodeInfo.ReleaseHash
	});
}
```

This way it's best effort to get the value and at most will parse 2x per file.

Reply:
Implemented the suggested changes, but was explicit instead of the null coalescing operator in case `sceneName` somehow is an empty string since FileNameBuilder performs the same check